### ISSUE TITLE
CATH inverse folding evaluation

### ISSUE THREAD

**Issue by alex-hh** [alex-hh/profam#338]
Want to be able to perform an Saprot style validation:

> We initially explored two intuitive approaches for protein structure modeling. The first approach
involves treating the predicted structures from AF2 as a graph and employing GNNs for modeling,
following (Yang et al., 2022). The second approach is to extract the distance and angle information
between pairwise residues from structures, incorporating it as a structure bias in a Transformer
(Vaswani et al., 2017) attention map. This approach was applied by Uni-Mol, ESMFold (Lin et al.,
2022) and Evoformer. We evaluate the two models4 using the MLM objective as it can support
both protein-level and residue-level tasks. It should be noted that the structure model in Uni-Mol,
ESMFold, and Evoformer were initially designed for specific tasks with different loss functions,
rather than being intended as general-purpose PLM. Therefore, it remains uncertain whether these
neural networks would be effective when trained with predicted structures using the MLM objective.
Through two exploratory experiments, we noted that training directly using predicted structures
yielded poor performance on the validation set containing real PDB structures (Figure 2). The
decrease in loss on predicted structures did not correspond to a decrease in loss on real structures.
This mismatch may be due to the fact that PLM has detected traces of AF2 predictions. Furthermore,
inferior results were reported in downstream tasks [Gym and deeploc] (Table 8). Despite a substantial loss decrease on
training data, these models failed to learn meaningful representations for downstream protein tasks.
